Flexa-Agents/Withdrawal-Root-Hash-Addition/src/agent.spec.ts

```diff
@@ -0,0 +1,100 @@
+import { Finding, FindingSeverity, FindingType, HandleTransaction, TransactionEvent } from "forta-agent";
+
+import { createAddress, TestTransactionEvent } from "forta-agent-tools";
+import { provideHandleTransaction, FLEXA_COLLATERAL_MANAGER_IFACE as IFACE } from "./agent";
+
+const factory: string = createAddress("0xa0");
+const unixTime = 1645787209;
+
+const createFinding = (txDate: number, txEvent: TransactionEvent) => {
+  const time = new Date(txDate * 1000);
+
+  return Finding.fromObject({
+    name: "Flexa withdrawal root event",
+    description: "A new withdrawal root hash is added to the active set",
+    alertId: "FLEXA-3",
+    severity: FindingSeverity.Info,
+    type: FindingType.Info,
+    protocol: "Flexa",
+    metadata: {
+      timestamp: txEvent.timestamp.toString(),
+      timeUTC: time.toUTCString()
+    }
+  });
+};
+
+describe("Flexa Withdrawal Root Hash Addition Agent Test Suite", () => {
+  const handler: HandleTransaction = provideHandleTransaction(factory);
+
+  it("should ignore empty transactions", async () => {
+    const tx: TransactionEvent = new TestTransactionEvent();
+
+    const findings = await handler(tx);
+    expect(findings).toStrictEqual([]);
+  });
+
+  it("should ignore same events from other contracts", async () => {
+    const log = IFACE.encodeEventLog(IFACE.getEvent("WithdrawalRootHashAddition"), [
+      "0x469d8ac2d6af2747f17ec8db2b783a24cf1ae0d08e5b4d16d908b7f90f63e90d",
+      "1"
+    ]);
+
+    const tx: TransactionEvent = new TestTransactionEvent()
+      .addAnonymousEventLog(createAddress("0xd4"), log.data, ...log.topics)
```

Comment:
It is correct, because of the topics, he already have the topics array that includes the sighash in `topic[0]`. The not Anonymous add the `topic[0]` to the topics you provide.